src/measure_theory/constructions/prod.lean

```diff
@@ -1102,4 +1102,43 @@ lemma set_integral_prod_mul {L : Type*} [is_R_or_C L]
   ∫ z in s ×ˢ t, f z.1 * g z.2 ∂(μ.prod ν) = (∫ x in s, f x ∂μ) * (∫ y in t, g y ∂ν) :=
 by simp only [← measure.prod_restrict s t, integrable_on, integral_prod_mul]
 
+/-! ### Marginals of a measure defined on a product -/
+
+namespace measure
+
+variables {ρ : measure (α × β)}
+
+/-- Marginal measure on `α` obtained from a measure `ρ` on `α × β`, defined by `ρ.map prod.fst`. -/
+noncomputable
+def fst (ρ : measure (α × β)) : measure α := ρ.map prod.fst
+
+lemma fst_apply {s : set α} (hs : measurable_set s) : ρ.fst s = ρ (prod.fst ⁻¹' s) :=
```

Comment:
is there a trick we can do similar to `measurable_embedding.map_apply` that gets rid of the assumption `hs`?

Reply:
I tried that, but without success so far. I'll merge as it is for now and come back to it later.